dts/bindings/adc/adc-comparator.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Comparator will monitor signal though ADC and based on configuration,
+    it will trigger a callback to user to inform signal criterion is met.
+
+    User is responsible for configuring comparator to establish threshold assert
+    value, comparison (less or equal, grater...) and callback handler.
```

Comment:
Since this binding has a compatible (`adc-comparator`), that makes it look like a generic binding which can be used in different contexts. I think some devicetree source examples would therefore be good to add here explaining how all of this works.

If this is not a generic binding, I wonder if a vendor prefix is missing? Or perhaps these properties are specific to the nuvoton part? Either way, something needs to be reworked here.

Reply:
> Since this binding has a compatible (`adc-comparator`), that makes it look like a generic binding which can be used in different contexts. I think some devicetree source examples would therefore be good to add here explaining how all of this works.
> 
> If this is not a generic binding, I wonder if a vendor prefix is missing? Or perhaps these properties are specific to the nuvoton part? Either way, something needs to be reworked here.

This is meant to be used as a base, see dts/bindings/adc/nuvoton,adc-cmp.yaml included on this commit. 
Should I remove compatible: "adc-comparator"?